File: src/de/hsh/dennis/model/NpcLogic/SpawnTimer.java
package de.hsh.dennis.model.NpcLogic;

public class SpawnTimer {

    //Uhr
    private double start = -1;

    public void start() {
        if (start == -1) {
            start = getCurrentSec();
        }
    }

    public double getCurrentTimeStamp() {
        start();
        return getCurrentSec() - start;
    }

    public double getCurrentSec() {
        return System.currentTimeMillis() / 1000d;
    }

    //Rundenzähler
    private double resetTime = -1;

    public double elabsedTime(){
        if(resetTime == -1){resetTimer();}
        return (System.currentTimeMillis() / 1000d) - resetTime;
    }

    public void resetTimer(){
        resetTime = System.currentTimeMillis() / 1000d;
    }
}

File: Games/Dennis/src/de/hsh/dennis/model/Spawn/SpawnTimer.kt
package de.hsh.dennis.model.Spawn

class SpawnTimer {
    //Uhr
    private var start = -1.0
    fun start() {
        if (start == -1.0) {
            start = currentSec
        }
    }

    val currentTimeStamp: Double
        get() {
            start()
            return currentSec - start
        }

    private val currentSec: Double
        get() = System.currentTimeMillis() / 1000.0

    //Rundenzähler
    private var resetTime = -1.0
    fun elabsedTime(): Double {
        if (resetTime == -1.0) {
            resetTimer()
        }
        return System.currentTimeMillis() / 1000.0 - resetTime
    }

    fun resetTimer() {
        resetTime = System.currentTimeMillis() / 1000.0
    }
}
